src/utils.h

```diff
@@ -24,14 +27,25 @@ SEXP r_parse_eval(const char* str, SEXP env);
 
 static inline
 SEXP r_new_environment(SEXP parent) {
+#if (R_VERSION >= R_Version(4, 1, 0))
+  return R_NewEnv(parent, FALSE, 0);
+#else
   SEXP env = Rf_allocSExp(ENVSXP);
   SET_ENCLOS(env, parent);
   return env;
+#endif
 }
 
 static inline
 SEXP r_env_get(SEXP env, SEXP sym) {
+#if (R_VERSION >= R_Version(4, 5, 0))
+  SEXP obj = PROTECT(R_getVarEx(sym, env, FALSE, r_unbound_sym));
```

Comment:
```suggestion
  SEXP obj = PROTECT(R_getVarEx(sym, env, FALSE, R_UnboundValue));
```

Reply:
I searched the R sources and Writing R Extensions and didn't find anything about `R_UnboundValue` being off limits? This would remove the need for `r_unbound_sym` and associated changes.

Reply:
Oh I see, so the whole point is to move away from use of `R_UnboundValue`. I guess that makes sense to prevent it from being accidentally returned to R (although I really don't see a problem here where we're using it entirely internally within C code).